repos/spack_repo/builtin/packages/kokkos_fft/package.py

```diff
@@ -0,0 +1,69 @@
+from spack.package import *
+
+
+class KokkosFft(CMakePackage):
+    """FFT interfaces for Kokkos C++ Performance Portability Programming EcoSystem"""
+
+    homepage = "https://github.com/kokkos/kokkos-fft"
+    url = "https://github.com/kokkos/kokkos-fft/archive/refs/tags/v0.3.0.tar.gz"
+
+    maintainers("cedricchevalier19", "tpadioleau", "yasahi-hpc")
+
+    license("Apache-2.0 WITH LLVM-exception OR MIT", checked_by="cedricchevalier19")
+
+    version("0.3.0", sha256="a13c423775afec5f9f79fa9a23dd6001d3d63bae9f4786b1e0cd3ed65b3993a3")
+
+    variant("host_backend", default="fftw-serial",
+            values=("fftw-serial", "fftw-openmp"),
+            multi=True,
+            description="Enable host backend")
+    variant("device_backend", default="none",
+            values=("none", "cufft", "hipfft", "onemkl"),
+            multi=False,
+            description="Enable device backend")
+    variant("tests", default=False, description="Enable tests")
+
+    depends_on("kokkos@4.4:4 +complex_align")
+    # kokkos-fft currently only supports compilation with the Kokkos nvcc wrapper
+    requires("^kokkos +cuda +wrapper", when="device_backend=cufft")
+    requires("^kokkos +rocm", when="device_backend=hipfft")
+    requires("^kokkos +sycl", when="device_backend=onemkl")
+    depends_on("googletest@1.15:1", when="+tests")
+
+    depends_on("fftw@3.3:3 ~mpi precision=float,double", when="host_backend=fftw-serial")
+    depends_on("fftw@3.3:3 ~mpi precision=float,double", when="host_backend=fftw-openmp")
+    requires("^fftw +openmp", when="host_backend=fftw-openmp")
+    depends_on("cuda@11:12", when="device_backend=cufft")
+    depends_on("hipfft@5.3:6", when="device_backend=hipfft")
+    depends_on("intel-oneapi-mkl@2023:2025", when="device_backend=onemkl")
+
+    def cmake_args(self):
+        args = [
+            self.define("KokkosFFT_ENABLE_INTERNAL_KOKKOS", False),
+            self.define_from_variant("KokkosFFT_ENABLE_TESTS", "tests"),
+            self.define("KokkosFFT_ENABLE_DOCS", False),
+            self.define("KokkosFFT_ENABLE_BENCHMARK", False),
+            self.define("KokkosFFT_ENABLE_EXAMPLES", False),
+        ]
+
+        if self.spec.satisfies("host_backend=fftw-serial") or self.spec.satisfies("host_backend=fftw-openmp"):
+            args.append(self.define("KokkosFFT_ENABLE_FFTW", "ON"))
+
+        if self.spec.satisfies("device_backend=cufft"):
+            args.append(self.define("KokkosFFT_ENABLE_CUFFT", "ON"))
+        if self.spec.satisfies("device_backend=hipfft"):
+            args.append(self.define("KokkosFFT_ENABLE_HIPFFT", "ON"))
+        if self.spec.satisfies("device_backend=onemkl"):
+            args.append(self.define("KokkosFFT_ENABLE_ONEMKL", "ON"))
+
+        if self.spec.satisfies("^kokkos+rocm"):
+            args.append(self.define("CMAKE_CXX_COMPILER", self.spec["hip"].hipcc))
+        else:
+            args.append(self.define("CMAKE_CXX_COMPILER", self.spec["kokkos"].kokkos_cxx))
```

Comment:
```suggestion
            args.append(self.define("CMAKE_CXX_COMPILER", self["kokkos"].kokkos_cxx))
```
`self.spec["kokkos"].kokkos_cxx` doesn't work with 1.0, it should be `self["kokkos"].kokkos_cxx`
what kokkos-kernels does: https://github.com/spack/spack-packages/blob/98e5bd13ea6d53481df3cfc0090e5999c052fafa/repos/spack_repo/builtin/packages/kokkos_kernels/package.py#L281-L284

Reply:
Thanks! I did not test it with Spack 1 actually :D

Reply:
done in 2a837689dd4692bc6f60b68e75a8d1d4da1d3dc7